app/javascript/react/components/capeform/NewCapeForm.js

```diff
@@ -0,0 +1,224 @@
+import _ from 'lodash';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import ErrorList from './ErrorList';
+
+const NewCapeForm = () => {
+  const defaultForm = {
+    name: '',
+    full_name: '',
+    gender: '',
+    affiliation: '',
+    intelligence: 1,
+    strength: 1,
+    speed: 1,
+    url: ''
+  };
+
+  const [form, setForm] = useState(defaultForm);
+  const [errors, setErrors] = useState({});
+  const [freshCape, setFreshCape] = useState(null);
+  const [redirect, setRedirect] = useState(false);
+  const [duplicateError, setDuplicateError] = useState('');
+
+  const validForSubmission = () => {
+    let submitErrors = {};
+    const requiredFields = Object.keys(form);
+
+    requiredFields.forEach(field => {
+      if (form[field].toString().trim() === '') {
+        submitErrors = {
+          ...submitErrors,
+          [field]: 'is blank'
+        };
+      }
+    });
+    setErrors(submitErrors);
+    return _.isEmpty(submitErrors);
+  };
+
+  const onFormChange = event => {
+    const key = event.currentTarget.id;
+    const value = event.currentTarget.value;
+    setForm({
+      ...form,
+      [key]: value
+    });
+  };
+
+  const clearForm = () => {
+    setForm(defaultForm);
+  };
+
+  const formSubmit = event => {
+    event.preventDefault();
+    if (validForSubmission()) {
+      event.preventDefault();
+      addSuperhero();
+    }
+  };
+
+  function addSuperhero() {
+    fetch('/api/v1/capes', {
+      credentials: 'same-origin',
+      method: 'POST',
+      body: JSON.stringify(form),
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      }
+    })
+      .then(response => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+            error = new Error(errorMessage);
+          throw error;
+        }
+      })
+      .then(response => response.json())
+      .then(body => {
+        if (body.id) {
+          setFreshCape(body.id);
+          setRedirect(true);
+        }
+        setDuplicateError(body.errors[0]);
+      })
+      .catch(error => console.error(`Error in fetch: ${error.message}`));
+  }
+
+  if (redirect) {
+    let path = `/superheroes/${freshCape}`;
+    return <Redirect to={path} />;
+  }
+
+  return (
+    <>
+      <div>
+        <h5>{duplicateError}</h5>
+      </div>
+
+      <form onSubmit={formSubmit}>
+        <ErrorList errors={errors} />
+        <h4>Create Your Custom Superhero:</h4>
+        <div className='grid-container'>
+          <div className='grid-x grid-padding-x'>
+            <div className='medium-6 cell'>
+              <label>
+                Alias
+                <input
+                  onChange={onFormChange}
+                  id='name'
+                  name='name'
+                  type='text'
+                  placeholder='Blobsweat'
```

Comment:
you should be consistent with either using placeholders for all input fields, or none. Placeholders can be great to show an example of the type of input you want, but a couple things to keep in mind: the example inputs should be a relevant example and also professional - consider changing Alias `Blobsweat` to a real superhero, `Superman`, and Affiliation to something like `Justice League` rather than Launch Academy